### PR TITLE
[CHK-1697] (feat): update iban debt position

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Update debt position iban in debt position update api

#### List of Changes

In update branch workflow add the iban update to the debt position.
Iban is retrieved from APIConf (named ibanClient in the code) and passed to the unique transfer of the debt position

#### Motivation and Context

This PR address a docs requirement about the update of debt position. When we invoke apiConfig to get company name, we need to use also company iban and update it in the transfer list item in payment option model.

#### How Has This Been Tested?

Junit test has been ran 

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/test/kotlin/it/pagopa/aca/service/AcaServiceTests.kt [hunk 1]
```diff
@@ -151,7 +152,8 @@ class AcaServiceTests {
         val responseCreate = AcaTestUtils.debitPositionModelResponse(iupd)
         /* preconditions */
         given(gpdClient.getDebtPosition(any(), any())).willReturn(Mono.just(responseGetPosition))
-        given(ibansClient.getIban(any(), any())).willReturn(Mono.just(Pair(ibanTest, companyName)))
+        given(ibansClient.getIban(any(), any()))
+            .willReturn(Mono.just(Pair(ibanTestUpdate, companyName)))
         given(gpdClient.updateDebtPosition(any(), any(), any()))
```
Comment (Fernando-Granato) [inline src/test/kotlin/it/pagopa/aca/service/AcaServiceTests.kt:157]: Use ArgumentCapture for check the value of any() contains iban updated 🙂
```suggestion
        given(gpdClient.updateDebtPosition(any(), any(), debitPositionToUpdateCapture.capture().))
        assertEquals(ibanUpdated, debitPositionToUpdateCapture.getValue().getIban())
```

**Review by Fernando-Granato — DISMISSED** [review]
🚀

**Review by Fernando-Granato — COMMENTED** [review]
Fix CR pipeline

File: src/test/kotlin/it/pagopa/aca/service/AcaServiceTests.kt [hunk 2]
```diff
@@ -29,12 +32,14 @@ class AcaServiceTests {
     private val ibansClient: IbansClient = mock()
     private val acaUtils = AcaUtils()
     private val acaService = AcaService(gpdClient, ibansClient, acaUtils)
+    @Captor private lateinit var paymentPositionModelDtoCaptor: ArgumentCaptor<PaymentPositionModelDto>
```
Comment (pietro-tota) [inline src/test/kotlin/it/pagopa/aca/service/AcaServiceTests.kt:35]:
```suggestion
    private val paymentPositionModelDtoCaptor: KArgumentCaptor<PaymentPositionModelDto> =
        argumentCaptor<PaymentPositionModelDto>()
```
🚲 use kotlin argument captor here

File: src/test/kotlin/it/pagopa/aca/service/AcaServiceTests.kt [hunk 3]
```diff
@@ -168,11 +174,13 @@ class AcaServiceTests {
...
                     requestCreatePosition,
                     iupd,
+                    ibanTestUpdate,
                     companyName
                 )
             )
         verify(gpdClient, Mockito.times(0)).createDebtPosition(any(), any())
         verify(gpdClient, Mockito.times(0))
             .invalidateDebtPosition(requestCreatePosition.paFiscalCode, iupd.value())
+        assertEquals(ibanTestUpdate, paymentPositionModelDtoCaptor.value.paymentOption?.get(0)?.transfer?.get(0)?.iban)
```
Comment (pietro-tota) [inline src/test/kotlin/it/pagopa/aca/service/AcaServiceTests.kt:184]:
```suggestion
        assertEquals(ibanTestUpdate, paymentPositionModelDtoCaptor.firstValue.paymentOption?.get(0)?.transfer?.get(0)?.iban)
```
and then here first value method from KArgumentMatcher class

**Review by pietro-tota — DISMISSED** [review]
Well done, just a comment. Remember to perform code format with spotless

File: src/test/kotlin/it/pagopa/aca/service/AcaServiceTests.kt (hunk 3 above)
Comment (ciuffagianluca) [reply src/test/kotlin/it/pagopa/aca/service/AcaServiceTests.kt]: Did it!

**Review by Fernando-Granato — APPROVED** [review]

**Pull request merged**